### PR TITLE
Miawong/add slow loader state to license api 

### PR DESCRIPTION
**Pull request by miaawong**
#### What this PR does / why we need it: This allows us to add a loading spinner if the data takes longer than .5 seconds to return. 
- Created a custom hook that run the` /license` query hook and return slowLoading state if the response takes longer than .5 seconds. 
- Created a query hook that queries `/license` endpoint
- Added axios to kots/web 
- Hook is called `useLicenseWithIntercept`
- Hook is not implemented yet and is not being used anywhere. 

#### Which issue(s) this PR fixes:

Fixes # [sc-50084]

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Does this PR require documentation?

NONE

### CODE REVIEW

File: web/src/components/apps/Dashboard.tsx [hunk 1]
```diff
@@ -38,6 +38,7 @@ import {
   Version,
 } from "@types";
 import { RouteComponentProps } from "react-router-dom";
+//import LicenseTester from "./LicenseTester";
```
Comment (miaawong) [inline web/src/components/apps/Dashboard.tsx:42]: can be deleted, wanted to make it easy to test

File: web/src/components/apps/Dashboard.tsx [hunk 2]
```diff
@@ -749,11 +751,24 @@ class Dashboard extends Component<Props, State> {
...
     return (
       <>
-        {!app && (
-          <div className="flex-column flex1 alignItems--center justifyContent--center">
-            <Loader size="60" />
-          </div>
-        )}
+        {!app ||
+          (this.state.slowLoader && (
+            <div
```
Comment (miaawong) [inline web/src/components/apps/Dashboard.tsx:756]: added slowLoader state, but this is always false for now. 
changed some of the styles so it only shows loader on the dashboard not the multi app sidebar

File: web/src/components/apps/Dashboard.tsx [hunk 3]
```diff
@@ -851,7 +867,15 @@ class Dashboard extends Component<Props, State> {
                       gettingAppLicenseErrMsg={
                         this.state.gettingAppLicenseErrMsg
                       }
-                    />
+                    >
+                      {/* leaving this here as an example: please delete later */}
+                      {/* <LicenseTester
```
Comment (miaawong) [inline web/src/components/apps/Dashboard.tsx:872]: component to test the useLicenseWithIntercept hook, please delete when it's unnecessary for you.

File: web/src/components/apps/LicenseTester.jsx [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  const { data } = licenseQuery;
+  const { license } = data || {};
+  const { entitlements } = license || [];
+
+  useEffect(() => {
+    setLoader(isSlowLoading);
+  }, [isSlowLoading, licenseQuery?.data]);
+
+  return (
+    <div>
```
Comment (miaawong) [inline web/src/components/apps/LicenseTester.jsx:18]: test component, can be deleted when it seems unnecessary. just wanted to leave an example here

File: web/webpack.config.js [hunk 5]
```diff
@@ -119,6 +119,12 @@ module.exports = function (env) {
               }
             }
           ]
+        }, 
+        {
+          test: /\.m?js/,
+          resolve: {
+              fullySpecified: false
+          }
```
Comment (miaawong) [inline web/webpack.config.js:127]: had to add this to get axios working
doesn't really make sense to me.

File: web/src/components/apps/LicenseTester.jsx (hunk 4 above)
Comment (rianfowler) [reply web/src/components/apps/LicenseTester.jsx:18]: thank you

File: web/src/features/App/api/useLicense.ts [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+  //   });
+  // };
+
+  let licenseQuery = useLicense(appSlug);
+
+  useEffect(() => {
+    axios.interceptors.request.use(
+      (x) => {
+        // not sure if this check is enough to make sure that it
+        // only happens on /license endpoint
```
Comment (rianfowler) [inline web/src/features/App/api/useLicense.ts:24]: so interceptors are globally applied? that's good to know

File: web/webpack.config.js (hunk 5 above)
Comment (rianfowler) [reply web/webpack.config.js:127]: axios must use .mjs files https://webpack.js.org/configuration/module/#resolvefullyspecified

**Review by rianfowler — APPROVED** [review]

File: web/src/features/App/api/useLicense.ts [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+          // TODO: handle unauthorized
+          console.log("an error occurred");
+        }
+        //TODO: handle other errors
+        console.log("Error => axios.interceptors.request", error);
+      }
+    );
+  }, []);
+  // return everything that comes with react query
+  return { licenseQuery, isSlowLoading };
```
Comment (rianfowler) [inline web/src/features/App/api/useLicense.ts:63]:
```suggestion
  return { ...licenseQuery, isSlowLoading };
```
This is how I think we should do it that way it's the same api for all of our api queries.

**Review by rianfowler — APPROVED** [review]

**Pull request merged**